src/Tasks/Microsoft.NET.Build.Tasks/CacheArtifactParser.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using System.Globalization;
+using System.Xml.Linq;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    //The following provides the logic to parse the output - artifact.xml
+    // generated by ComposeCache
+    internal class CacheArtifactParser
+    {
+        internal static HashSet<PackageInfo> Parse(string filterProjFilesList)
+        {
+            var listofFilterFiles = filterProjFilesList.Split(';');
+            var listofPackages = new HashSet<PackageInfo>();
+
+            foreach (var filterFile in listofFilterFiles)
+            {
+                var doc = XDocument.Load(filterFile);
+
+                foreach (var pkginfo in doc.Elements("Project").Elements("ItemGroup").Elements("PackageReference"))
```

Comment:
For Xml safety reasons, you should probably prepend the document's namespace.  Just in case someone adds the msbuild namespace to the document.

See [https://github.com/dotnet/sdk/blob/master/test/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs#L100-L104](https://github.com/dotnet/sdk/blob/master/test/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs#L100-L104) as an example.